src/common/xio_connection.c

```diff
@@ -1751,6 +1751,9 @@ static void xio_fin_req_timeout(void *data)
 	/* flush all messages back to user */
 	xio_connection_notify_msgs_flush(connection);
 
+	if (connection->nexus)
+		xio_nexus_close(connection->nexus, &connection->session->observer);
```

Comment:
OK, I see it - done in client/server observer in due to the `XIO_NEXUS_EVENT_CLOSED` notification.

BTW, this is done in `xio_nexus_close()` under the condition `if (observer)`. Is it at all possible that `observer==NULL` having been initialized in `xio_nexus_create()`?